tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala

```diff
@@ -17,20 +17,26 @@
 
 package com.spotify.featran
 
-import org.tensorflow.example.{Example, Features}
-import org.tensorflow.{example => tf}
 import _root_.java.util.regex.Pattern
 
 import com.spotify.featran.transformers.{MDLRecord, WeightedLabel}
+import org.tensorflow.example.{Example, Features}
+import org.tensorflow.{example => tf}
 import shapeless.datatype.tensorflow.TensorFlowType
 
-case class NamedTFFeature(name: String, f: tf.Feature)
-
 package object tensorflow {
-  private val FeatureNameNormalization = Pattern.compile("[^A-Za-z0-9_]")
+
+  private[this] object FeatureNameNormalization {
+    private[this] val NamePattern = Pattern.compile("[^A-Za-z0-9_]")
+
+    val normalize: String => String = fn =>
+      NamePattern.matcher(fn).replaceAll("_")
```

Comment:
Do we want to add memoization here also?

Reply:
@richwhitjr yup I guess it can improve things a little bit